tensorflow/src/TensorFlow/Types.hs

```diff
@@ -124,6 +126,11 @@ import qualified TensorFlow.Internal.FFI as FFI
 
 type ResourceHandle = ResourceHandleProto
 
+-- | Dynamic type.
+-- TensorFlow variants aren't supported yet. This type acts a placeholder to
+-- simplify op generation.
+data Variant = Variant
```

Comment:
Is void type at all plausible here? I.e. 
data Variant

Reply:
Yeah. I've removed the data constructor.